includes/chart.js

```diff
@@ -0,0 +1,330 @@
+                document.getElementById("clicked-node").textContent = "Clicked Node: " + `${row.id}_${template_node.id}`;
+            });
+
+            // Make the group draggable
+            group.addEventListener('mousedown', function (event) {
+                var offsetX, offsetY;
+
+                if (newNode.tagName === 'ellipse') {
+                    offsetX = event.clientX - parseFloat(newNode.getAttribute("cx"));
+                    offsetY = event.clientY - parseFloat(newNode.getAttribute("cy"));
+                } else if (newNode.tagName === 'polygon') {
+                    var bbox = newNode.getBoundingClientRect();
+                    offsetX = event.clientX - bbox.left;
+                    offsetY = event.clientY - bbox.top;
+                } else if (newNode.tagName === 'rect') {
+                    offsetX = event.clientX - parseFloat(newNode.getAttribute("x"));
+                    offsetY = event.clientY - parseFloat(newNode.getAttribute("y"));
+                }
+
+                function moveNode(event) {
+                    if (newNode.tagName === 'ellipse') {
+                        newNode.setAttribute("cx", event.clientX - offsetX);
+                        newNode.setAttribute("cy", event.clientY - offsetY);
+                        // Update text position for ellipse
+                        text.setAttribute("x", parseFloat(newNode.getAttribute("cx")));
+                        text.setAttribute("y", parseFloat(newNode.getAttribute("cy")));
+                    } else if (newNode.tagName === 'polygon') {
+                        var bbox = newNode.getBoundingClientRect();
+                        var deltaX = event.clientX - offsetX - bbox.x;
+                        var deltaY = event.clientY - offsetY - bbox.y;
+                        // Update polygon points
+                        var points = newNode.points;
+                        var numPoints = points.numberOfItems;
+                        for (var i = 0; i < numPoints; i++) {
+                            points.getItem(i).x += deltaX;
+                            points.getItem(i).y += deltaY;
+                        }
+                        // Update text position for polygon
+                        text.setAttribute("x", parseFloat(text.getAttribute("x")) + deltaX);
+                        text.setAttribute("y", parseFloat(text.getAttribute("y")) + deltaY);
+
+                    } else if (newNode.tagName === 'rect') {
+                        newNode.setAttribute("x", event.clientX - offsetX);
+                        newNode.setAttribute("y", event.clientY - offsetY);
+                        // Update text position for rectangle
+                        text.setAttribute("x", parseFloat(newNode.getAttribute("x")) + parseFloat(newNode.getAttribute("width")) / 2);
+                        text.setAttribute("y", parseFloat(newNode.getAttribute("y")) + parseFloat(newNode.getAttribute("height")) / 2);
```

Comment:
Do you have a linter for line length? More than 100something chars is quite long.

Reply:
Not yet, this is an open issue #18